pkg/memstore/memstore.go

```diff
@@ -0,0 +1,63 @@
+package memstore
+
+import (
+	"encoding/json"
+	"sync"
+)
+
+type Bucket interface {
+	Set(string, interface{})
+	Get(string) interface{}
+	MarshallJSON() ([]byte, error)
+}
+
+type Store interface {
```

Comment:
`memstore/memstore.go` sound a bit redundant, maybe we can rename it `memstore/store.go` ?